tests/functional/test_contract_instance.py

```diff
@@ -638,3 +640,25 @@ def test_obj_list_as_struct_array_input(contract_instance, owner, data_object):
 def test_dict_list_as_struct_array_input(contract_instance, owner):
     data = {"a": owner, "b": HexBytes(123), "c": "GETS IGNORED"}
     assert contract_instance.setStructArray([data, data]) is None
+
+
+def test_custom_error(error_contract, not_owner):
+    contract = error_contract
+    unauthorized = contract.Unauthorized
+    assert issubclass(unauthorized, CustomError)
+
+    with pytest.raises(contract.Unauthorized) as err:
```

Comment:
@fubuloubu Look at this! This means everything you think it means. I think we're good now, at least with this part?

Reply:
Yes, this seems great!